File: api/schemas/dmarc_report_detail_tables/dmarc_report_detail_tables.py
import graphene

from scalars.year import Year
from schemas.shared_structures.detail_tables import DetailTables


class DmarcReportDetailTables(graphene.ObjectType):
    """
    GraphQL object for returning data for dmarc report tables
    """

    month = graphene.String(description="The month in which the data is relevant to.")
    year = Year(description="The year in which the relevant to.")
    detail_tables = graphene.Field(
        lambda: DetailTables,
        description="The details used in creating tables for the dmarc report " "page",
    )

File: api-js/src/types/dmarc-report/dmarc-report-detail-tables.js
const { GraphQLString, GraphQLObjectType } = require('graphql')
const { Year } = require('../../scalars')
const { detailTableType } = require('./detail-tables')

const dmarcReportDetailedTablesType = new GraphQLObjectType({
  name: 'DmarcReportDetailedTables',
  description: `GraphQL object for returning data for dmarc report tables.`,
  fields: () => ({
    month: {
      type: GraphQLString,
      description: `The month in which the table data is relevant to.`,
      resolve: async () => {},
    },
    year: {
      type: Year,
      description: `The year in which the data is relevant to.`,
      resolve: async () => {},
    },
    detailTables: {
      type: detailTableType,
      description: `The details used in creating tables for the dmarc report page.`,
      resolve: async () => {},
    },
  }),
})

module.exports = {
  dmarcReportDetailedTablesType,
}
